extide/gradle/test/unit/src/org/netbeans/modules/gradle/NbGradleProjectImplTest.java

```diff
@@ -220,18 +244,18 @@ public void testIncreaseAimedQualityChangesProject() throws Exception {
      * the evaluated state does not exist.
      */
     public void testEvaluateTrustedDoesNotExecuteScript() throws Exception {
-        Project prj = createProject();
         NbGradleProject ngp = NbGradleProject.get(prj);
         ProjectTrust.getDefault().trustProject(prj);
         // initializes the project
         assertTrue(ngp.getQuality().worseThan(NbGradleProject.Quality.EVALUATED));
-        
+        assertHasNoConnection(prj);
         NbGradleProjectImpl prjImpl = prj.getLookup().lookup(NbGradleProjectImpl.class);
         GradleProject curProject = prjImpl.getGradleProject();
         prjImpl.setAimedQuality(Quality.EVALUATED);
         GradleProject newProject = prjImpl.getGradleProject();
         assertTrue(newProject.getQuality().worseThan(NbGradleProject.Quality.EVALUATED));
         assertSame(newProject, curProject);
+        assertHasNoConnection(prj);
```

Comment:
What kinds of actions can cause the project to have a connection? I assume there is at least one test that causes hasConnection to become true, since it was necessary to add logic to close the connection in tearDown? If the latter is true, does this mean we are now depending on the various test methods to execute in a specific order, so that the connection is only made later?

Reply:
Well certain operation shall not create any connections. These are just checkpoints, to fail the test if that would ever happen.